build_system/src/utils.rs

```diff
@@ -238,3 +299,68 @@ where
     }
     Ok(())
 }
+
+pub fn split_args(args: &str) -> Result<Vec<String>, String> {
+    let mut out = Vec::new();
+    let mut start = 0;
+    let args = args.trim();
+    let mut iter = args.char_indices().peekable();
+
+    while iter.peek().is_some() {
```

Comment:
Do you really need this outer loop? If so, can you add a comment to explain why?

Reply:
No it appears to be useless. Good catch!